pkg/api/dictionaryinfo.go

```diff
@@ -0,0 +1,338 @@
+// pkg/api/dictionaryinfo.go
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// DictionaryInfo is the Fastly API response for the /info endpoint.
+type DictionaryInfo struct {
+	Digest      string `json:"digest"`
+	ItemCount   int64  `json:"item_count"`
+	LastUpdated string `json:"last_updated"` // may be RFC3339 or other formats; can be empty
+}
+
+type svc struct {
+	ID       string `json:"id"`
+	Name     string `json:"name"`
+	Versions []struct {
+		Number int  `json:"number"`
+		Active bool `json:"active"`
+	} `json:"versions"`
+}
+
+type dict struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+// dictionary item model (subset)
+type dictItem struct {
+	ItemKey   string `json:"item_key"`
+	ItemValue string `json:"item_value"`
+	UpdatedAt string `json:"updated_at"`
+	CreatedAt string `json:"created_at"`
+}
+
+type DictionaryInfoCache struct {
+	client       *http.Client
+	token        string
+	logger       log.Logger
+	serviceCache *ServiceCache
+	enabled      bool
+
+	// Cached snapshot used by the collector to avoid network I/O on scrape.
+	snapshot []metricRow
+}
+
+type metricRow struct {
+	ServiceID      string
+	ServiceName    string
+	Version        int
+	DictionaryID   string
+	DictionaryName string
+	Digest         string
+	ItemCount      float64
+	LastUpdatedTS  float64
+}
+
+func NewDictionaryInfoCache(client *http.Client, token string, logger log.Logger, serviceCache *ServiceCache, enabled bool) *DictionaryInfoCache {
+	if logger == nil {
+		logger = log.NewNopLogger()
+	}
+	return &DictionaryInfoCache{
+		client:       client,
+		token:        token,
+		logger:       log.With(logger, "component", "dictionary-info"),
+		serviceCache: serviceCache,
+		enabled:      enabled,
+	}
+}
+
+func (c *DictionaryInfoCache) Enabled() bool { return c.enabled }
+
+// parse multiple time layouts commonly seen in Fastly APIs
+func parseFastlyTime(s string) (float64, bool) {
+	if s == "" {
+		return 0, false
+	}
+	layouts := []string{
+		time.RFC3339Nano,
+		time.RFC3339,
+		"2006-01-02 15:04:05",       // "2025-10-21 18:48:00"
+		"2006-01-02T15:04:05",       // without zone
+		"2006-01-02 15:04:05Z07:00", // space + zone
+	}
+	for _, layout := range layouts {
+		if t, err := time.Parse(layout, s); err == nil {
+			return float64(t.Unix()), true
+		}
+	}
+	return 0, false
+}
```

Comment:
This should be moved to the bottom and only need to support two layouts:

```
time.RFC3339,
"2006-01-02 15:04:05",
```

And because you don't need have the `bool` to fallback to listing dictionary items you can also remove the bool from the return signature.